lib/internal/modules/esm/module_job.js

```diff
@@ -151,7 +151,17 @@ class ModuleJob {
     await this.instantiate();
     const timeout = -1;
     const breakOnSigint = false;
-    await this.module.evaluate(timeout, breakOnSigint);
+    try {
+      await this.module.evaluate(timeout, breakOnSigint);
+    } catch (e) {
+      if (e.name === 'ReferenceError' &&
+          e.message === 'require is not defined') {
+        e.message += ' in ES module scope. You can use `await import` instead' +
+                     ', or use `.cjs` extension to load the file as CommonJS ' +
```

Comment:
Most people should be using the .js extension; it’s only if they out type:module in package.json that .cjs would be required. Can we clarify that too?

Reply:
Good call, I've added a test on the module URL to have the message about the extension only if the module is a file using `.js` extension.